modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -495,8 +505,8 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 					k++;
 				}
 
-				if (str[k] == ':') {
-					// has type hint
+				if (str[k] == ':' && in_function_arg_dicts == 0) {
+					// Has type hint
```

Comment:
Add punctuation.

Reply:
just a period at the end like - 'Has type hint.' is that right? or do you want a dash in there too. 'Has type - hint.' ?

Reply:
yes, to be like the other comments in this file, all has punctuation at the end.